src/internal/observable/combineLatest.ts

```diff
@@ -11,322 +11,45 @@ import { popResultSelector, popScheduler } from '../util/args';
-
-/** @deprecated Pass arguments in a single array instead `combineLatest([a, b, c])` */
-export function combineLatest<O extends ObservableInput<any>, R>(
-  ...observables: Array<ObservableInput<any> | ((...values: Array<any>) => R)>
 ): Observable<R>;
 
-/** @deprecated resultSelector no longer supported, pipe to map instead */
-export function combineLatest<O extends ObservableInput<any>, R>(
-  array: O[],
-  resultSelector: (...values: ObservedValueOf<O>[]) => R,
+/** @deprecated The scheduler argument is deprecated, use scheduled and combineAll. Details: https://rxjs.dev/deprecations/scheduler-argument */
+export function combineLatest<A extends readonly unknown[], R>(
+  sources: readonly [...ObservableInputTuple<A>],
+  resultSelector: (...args: A) => R,
   scheduler?: SchedulerLike
 ): Observable<R>;
 
-/** @deprecated Passing a scheduler here is deprecated, use {@link subscribeOn} and/or {@link observeOn} instead */
-export function combineLatest<O extends ObservableInput<any>>(...observables: Array<O | SchedulerLike>): Observable<any[]>;
+export function combineLatest<A extends readonly unknown[]>(sources: readonly [...ObservableInputTuple<A>]): Observable<A>;
 
-/** @deprecated Passing a scheduler here is deprecated, use {@link subscribeOn} and/or {@link observeOn} instead */
-export function combineLatest<O extends ObservableInput<any>, R>(
-  ...observables: Array<O | ((...values: ObservedValueOf<O>[]) => R) | SchedulerLike>
-): Observable<R>;
+/** @deprecated Use the version that takes an array of Observables instead */
+export function combineLatest<T, R>(source: ObservableInput<T>, resultSelector: (value: T) => R, scheduler?: SchedulerLike): Observable<R>;
+/** @deprecated Use the version that takes an array of Observables instead */
+export function combineLatest<T>(source: ObservableInput<T>, scheduler: SchedulerLike): Observable<[T]>;
+/** @deprecated Use the version that takes an array of Observables instead */
+export function combineLatest<T>(source: ObservableInput<T>): Observable<[T]>;
 
-/** @deprecated Passing a scheduler here is deprecated, use {@link subscribeOn} and/or {@link observeOn} instead */
-export function combineLatest<R>(
-  ...observables: Array<ObservableInput<any> | ((...values: Array<any>) => R) | SchedulerLike>
+/** @deprecated Use the version that takes an array of Observables instead */
+export function combineLatest<A extends readonly unknown[], R>(...sources: [...ObservableInputTuple<A>, (...args: A) => R]): Observable<R>;
+/** @deprecated Use the version that takes an array of Observables instead */
+export function combineLatest<A extends readonly unknown[], R>(
+  ...sources: [...ObservableInputTuple<A>, (...args: A) => R, SchedulerLike]
 ): Observable<R>;
+/** @deprecated Use the version that takes an array of Observables instead */
+export function combineLatest<A extends readonly unknown[]>(...sources: [...ObservableInputTuple<A>, SchedulerLike]): Observable<A>;
```

Comment:
question: Why does this use an approach that differs from what's done with `merge`? Does this behave a little better? Are there tests to show this?